packages/browser/__tests__/Session.spec.ts

```diff
@@ -260,6 +260,32 @@ describe("Session", () => {
       );
     });
 
+    it("logs the session out when its tokens expire", async () => {
+      jest.useFakeTimers();
+      const MOCK_TIMESTAMP = 10000;
+      jest.spyOn(Date, "now").mockReturnValueOnce(MOCK_TIMESTAMP);
+      const clientAuthentication = mockClientAuthentication();
+      const incomingRedirectHandler = jest.spyOn(
+        clientAuthentication,
+        "handleIncomingRedirect"
+      );
+      incomingRedirectHandler.mockResolvedValueOnce({
+        isLoggedIn: true,
+        sessionId: "Arbitrary session ID",
+        expirationDate: MOCK_TIMESTAMP + 1337,
+      });
+      const mySession = new Session({ clientAuthentication });
+      await mySession.handleIncomingRedirect("https://some.url");
+      expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1337);
+      expect(mySession.info.isLoggedIn).toBe(true);
+
+      const expireTimeout = ((setTimeout as unknown) as jest.Mock<
+        typeof setTimeout
+      >).mock.calls[0][0];
+      await expireTimeout();
```

Comment:
I'm discovering this part of the jest API, so this is an educational question for me: why couldn't `jest.runAllTimers();` be used there ?

Reply:
It was new for me too :) But the reason I didn't is because the function we pass to `setTimeout` returns a Promise which, in regular code, will be ignored, but is useful in the test to tell us when logging out actually succeeded (because that's async as well). So now the `expect` on the next line will actually succeed, rather than being called before `logout` is done.

But this is a reasonable question; I'll add a comment to mention this as well.

Reply:
39dec622